packages/catalog-client/src/types.ts

```diff
@@ -17,7 +17,7 @@
 import { Entity, EntityName, Location } from '@backstage/catalog-model';
 
 export type CatalogEntitiesRequest = {
-  filter?: Record<string, string | string[]> | undefined;
+  filter?: Record<string, string | string[]>[] | undefined;
```

Comment:
you probably want to make this support both the old type and the array form of the old type (and undefined). That way, callers don't need to change everywhere. Then your actual code can check `Array.isArray(filter) ? filter : [filter]` or do `[filter].flat()` or so, to easily handle both variants.